specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json

```diff
@@ -1842,6 +1842,22 @@
         "sourceVault"
       ],
       "description": "Key Vault Key Url and vault id of KeK, KeK is optional and when provided is used to unwrap the encryptionKey"
+    },
+	"KeyForDiskEncryptionSet": {
+      "properties": {
+        "sourceVault": {
+          "$ref": "#/definitions/SourceVault",
+          "description": "Resource id of the KeyVault containing the key or secret. This property is optional."
```

Comment:
This property is only there for backwards compatibility and there is no need to specify it in any case, right? If so, say so in the comment.

Reply:
It is not here only for backwards compatibility. Portal uses this field to perform Access validation and grant access if customer wants to. 
Also it is useful to keep track of where the keyvault key resides. So ideally someone should skip this only if they want to have keyvault form another subscription

Reply:
ok, so we recommend customers specify it if it is in the same subscription but not otherwise? Will it work if it is in a different sub? What is the guidance we give about when to specify the key vault?

Reply:
It will fail if they try to give a vault in a different sub.
Guidance should be to skip/remove KeyVault resourceId from a DES if the vault is in a different subscription from the DES.
Should we include this as a comment in the swagger ?

Reply:
yes, we should.